src/dune_engine/sandbox.ml

```diff
@@ -69,8 +77,18 @@ let link_function ~(mode : Sandbox_mode.some) =
       | true -> win32_error mode
       | false -> fun src dst -> Io.portable_hardlink ~src ~dst))
 
-let link_deps t ~mode ~deps =
-  let link = Staged.unstage (link_function ~mode) in
+let copy_and_make_writable src dst =
```

Comment:
One possible issue here is that the action might rely on the fact that some source files don't have write permissions. (For example, maybe it's a test that ensures that source files have correct permissions.) 

By removing write permissions from everything, we are potentially breaking such actions. Also, aren't we also going to rewrite the corresponding files in the source tree, i.e. make them all writable (since we would detect this change)?

Reply:
Even if we don't want to improve this behaviour now, we might want to add a test that demonstrates it.

Reply:
Agreed. I don't have a good solution to offer here. I suggest we leave it as a known issue and see if it's a problem in practice.

Reply:
I added a test to document this behavior.